job-server/src/spark.jobserver/JobInfoActor.scala

```diff
@@ -33,18 +33,19 @@ class JobInfoActor(jobDao: JobDAO, contextSupervisor: ActorRef) extends Instrume
 
   override def wrappedReceive: Receive = {
     case GetJobStatuses(limit) =>
-      sender ! jobDao.getJobInfos(limit.get)
+      val jobInfos = Await.result(jobDao.getJobInfos(limit.get), 60 seconds)
```

Comment:
This Await is not necessary and just adds more synchronous code with timeouts that will result in a TimeoutError later.  Instead, let's make it reactive:

    val originator = sender
    jobDao.getJobInfos(limit.get).foreach(originator ! _)

This sends the result to the sender whenever it is ready, no need for a timeout, and no need to block the thread of an actor which is a bad design pattern.